test-clients/src/main/java/com/hedera/services/bdd/spec/queries/crypto/ExpectedTokenRel.java

```diff
@@ -50,15 +54,15 @@ public static void assertNoUnexpectedRels(
 			List<String> expectedAbsent,
 			List<TokenRelationship> actualRels,
 			HapiApiSpec spec
-	) {
+	) throws Throwable {
 		for (String unexpectedToken : expectedAbsent) {
 			for (TokenRelationship actualRel : actualRels) {
 				var unexpectedId = spec.registry().getTokenID(unexpectedToken);
 				if (actualRel.getTokenId().equals(unexpectedId)) {
-					Assert.fail(String.format(
-							"Account '%s' should have had no relationship with token '%s'!",
-							account,
-							unexpectedToken));
+					String errMsg = String.format("Account '%s' should have had no relationship with token '%s'!",
+							account,unexpectedToken);
+					log.error(errMsg);
+					throw new Exception(errMsg);
```

Comment:
Same here, throw a specific exception. I think originally `Assert.fail` was used for this purpose without going through the hassle of creating new specific exception. And it was better than throwing a generic exception here

Reply:
Here I will try to answer this and the another question above together. First, why only some places are touched. The reason is that the original Assert.fail() works well for many occasions. For example, when we are constructing a request (query or  transaction) to send to services, we know that we are trying to construct good (error free) requests to pass all the checks hopefully (unless we on purposely construct requests for negative functional test cases). So we don't expect any error to happen in those occasions, that's why I still keep original assertions untouched. For the situations when we wait for the results back, things are different. For the exact same good request, it may succeed this time, but fails in next run, and we should take these different outcome as normal in perf regression test, hence my changes are and should only focus on these places. Second,  replacing Assert.fail() is to avoid tons of stack trace in perf tests. Especially for long duration test, the log file can be excessive if not curbed. It's true we can still use Assert.fail(), but to avoid excessive stack trace, I need to do another catch in original exception's catch clause. That I think it's not necessary; Third, why re-throwing a new exception. The reason is if not, the already failed op would still go thru all following processing steps and we have to add logic to handle it. So re-throwing an exception will bypass these actions. Yeah, this issue takes much more time to decide which places to change than actually do the changes.